EIPS/abstraction.md

```diff
@@ -0,0 +1,95 @@
+### Preamble
+
+    EIP: <to be assigned>
+    Title: Abstraction of transaction origin and signature
+    Author: Vitalik Buterin
+    Type: Standard Track
+    Category: Core
+    Status: Draft
+    Created: 2017-02-10
+    
+### Summary
+
+Implements a set of changes that serve the combined purpose of "abstracting out" signature verification and nonce checking, allowing users to create "account contracts" that perform any desired signature/nonce checks instead of using the mechanism that is currently hard-coded into transaction processing.
+
+### Parameters
+
+* METROPOLIS_FORK_BLKNUM: TBD
+* CHAIN_ID: same as used for EIP 155 (ie. 1 for mainnet, 3 for testnet)
+* NULL_SENDER: 2**160 - 1
+
+### Specification
+
+If `block.number >= METROPOLIS_FORK_BLKNUM`, then:
+1. If the signature of a transaction is `(CHAIN_ID, 0, 0)` (ie. `r = s = 0`, `v = CHAIN_ID`), then treat it as valid and set the sender address to `NULL_SENDER`
+2. Transactions of this form MUST have gasprice = 0, nonce = 0, value = 0, and do NOT increment the nonce of account NULL_SENDER.
+3. Set the address of any contract created through a creation transaction (whether of the traditional or the EIP86 variety) to equal `sha3(NULL_SENDER + sha3(init code)) % 2**160`, where `+` represents concatenation, replacing the earlier address formula of `sha3(rlp.encode([sender, nonce]))`
+4. Create a new opcode at `0xfb`, `CREATE2`, with 4 stack arguments (value, salt, mem_start, mem_size) which sets the creation address to `sha3(sender + salt + sha3(init code)) % 2**160`, where `salt` is always represented as a 32-byte value.
+5. Add to _all_ contract creation operations, including transactions and opcodes, the rule that if a contract at that address already exists and has non-empty code OR non-empty nonce, the operation fails and returns 0 as if the init code had run out of gas. If an account has empty code and nonce but nonempty balance, the creation operation may still succeed.
```

Comment:
There is no gas here. The intention is that the contract you call will have `block.coinbase.send(fee)` somewhere which will incentivize miners to include it in a block. 

However miners will only include transactions that pay them. Could this result in valid transactions that never get executed as the payment fails and thus no miner wants to include them? 

This could lead to alot of over heard for miners dealing with these transactions that will never get processed but are still valid.